frontend/src/Judge/pages/Vote.js

```diff
@@ -0,0 +1,157 @@
+    }).isRequired,
+    handlePrevious: PropTypes.func.isRequired,
+    handleNext: PropTypes.func.isRequired,
+    submission: PropTypes.object,
+    previous: PropTypes.shape({
+      id: PropTypes.number
+    }),
+    next: PropTypes.shape({
+      id: PropTypes.number
+    })
+  }
+
+  static defaultProps = {
+    submission: null
+  }
+
+  componentDidMount () {
+    // TODO:
+    // a) If we're visiting this page for the first time (/vote)
+    //   1. fetch all entries for the show we're voting on
+    //   2. create the shuffle order
+    //   3. identify the first unvoted entry; set the 'viewing' state properly
+    //   4. go to the route /show/<show_id>/vote?on=<entry_id> so that the page will re-render populated w/ the entry
+    // b) If we're visiting a specific entry for the first time (/vote?on=<entry_id>) (i.e. refresh)
+    //   1. fetch all entries for the show we're voting on
+    //   2. create the shuffle order
+    //   3. update the 'viewing' state based on the entry in the query param
+    //   4. render the entry in the query param
+    // c) If we're returning to this page (/vote) (i.e. entries and shuffle order exist)
+    //   1. identify the first unvoted entry; set the 'viewing' state properly
+    //   2. go to the route /show/<show_id>/vote?on=<entry_id> so that the page will re-render populated w/ the entry
+    // d) If we're returning to a specific entry on this page (/vote?on=<entry_id>) (i.e. from a) 4. or from review page)
+    //   1. update the 'viewing' state based on the entry in the query param (if they don't already match)
+    //   2. render the entry in the query param
+  }
+
+  render () {
+    const { show, handleNext, handlePrevious, submission, previous, next } = this.props
+
+    return (
+      <Container fluid>
+        <Row>
+          <Col xs='1'>
+            { previous && previous.id
+              ? (
+                <Link to={`/show/${show.id}/vote?on=${previous.id}`}>
+                  <Previous onClick={handlePrevious}>
+                    <FaChevronLeft size='4em' />
+                  </Previous>
+                </Link>
+              ) : null }
+          </Col>
+          <Col xs='10' style={{ minHeight: '500px' }}>
+            <SubmissionContainer>
+              {submission ? <Submission submission={submission} /> : null}
+            </SubmissionContainer>
+          </Col>
+          <Col xs='1'>
+            { next && next.id
+              ? (
+                <Link to={`/show/${show.id}/vote?on=${next.id}`}>
+                  <Next onClick={handleNext}>
+                    <FaChevronRight size='4em' />
+                  </Next>
+                </Link>
+              ) : null }
+          </Col>
+        </Row>
+      </Container>
+    )
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
+  const showId = ownProps.match.params.id
+  const { order, viewing } = state.judge.queues[showId]
+  const submissions = state.judge.submissions
+
+  const obj = {
```

Comment:
That's the point? The variable name isn't meant to be significant here. Normally we just return a plain object here – I did it this way to be explicit that the `previous` and `next` properties are added dynamically.

If you really want me to (to avoid bikeshedding), I can change it to:
```js
return {
  show: {
    id: showId
  },
  submission: submissions[order[viewing]],
  previous: viewing > 0 ? { id: order[viewing - 1] } : undefined
  next: viewing < order.length ? { id: order[viewing + 1] } : undefined
}
```

Reply:
I mean, I’m okay with it as is I suppose. I just found it a bit jarring to read.